strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,254 @@
+whereas the scheme for subgraphs uses red nodes and edges.
+
+.. autosummary::
+    plot
+    plot_subgraph
+
+Code details
+^^^^^^^^^^^^
+"""
+from typing import Optional, Tuple
+
+import networkx as nx
+
+
+def _node_coords(graph: nx.Graph, l: dict) -> Tuple:
+    """Converts coordinates for the graph edges for plotting purposes.
+
+    Args:
+        graph (nx.Graph): input graph
+        l (dict[int, float]): dictionary of nodes and their respective coordinates, can be
+        generated using a NetworkX `layout <https://networkx.github.io/documentation/latest/
+        reference/drawing.html#module-networkx.drawing.layout>`__
+
+    Returns:
+         Dict[str, list]: lists of x and y coordinates accessed as keys of a dictionary
+    """
+    n_x = []
+    n_y = []
+
+    for n in graph.nodes():
+        n_x.append(l[n][0])
+        n_y.append(l[n][1])
+
+    return {"x": n_x, "y": n_y}
+
+
+def _edge_coords(graph: nx.Graph, l: dict) -> dict:
+    """Converts coordinates for the graph edges for plotting purposes.
+
+        Args:
+            graph (nx.Graph): input graph
+            l (dict): dictionary of nodes and their respective coordinates
+
+        Returns:
+             dict: x and y coordinates for beginning and end of each edge. `None` is placed as a
+             separator between pairs of nodes/edges.
+        """
+    e_x = []
+    e_y = []
+
+    for e in graph.edges():
+
+        start_x, start_y = l[e[0]]
+        end_x, end_y = l[e[1]]
+
+        e_x.append(start_x)
+        e_x.append(end_x)
+
+        e_y.append(start_y)
+        e_y.append(end_y)
+
+        e_x.append(None)
+        e_y.append(None)
+
+    return {"x": e_x, "y": e_y}
+
+
+GREEN = "#3e9651"
+RED = "#cc2529"
+LIGHT_GREY = "#CDCDCD"
+
+graph_node_colour = GREEN
+graph_edge_colour = LIGHT_GREY
+subgraph_node_colour = RED
+subgraph_edge_colour = RED
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
+def plot(graph: nx.Graph, subgraph: Optional[list] = None, size: float = 500) -> None:
+    """Creates a plotly plot of the input graph.
+
+    This function can plot just the input graph or the graph with a specified subgraph highlighted.
+
+    **Example usage**:
+
+    >>> graph = nx.complete_graph(10)
+    >>> fig = plot(graph, [0, 1, 2, 3])
+    >>> fig.show()
+
+    .. image:: ../../_static/complete_graph.png
+       :width: 40%
+       :align: center
+       :target: javascript:void(0);
+
+    Args:
+        graph (nx.Graph): input graph
+        subgraph (list): optional list of nodes comprising the subgraph to highlight
+        size (float): size of the plot in pixels, rendered in a square layout
+
+    Returns:
+         Figure: Plotly figure for graph and optionally highlighted subgraph
+    """
+    try:
+        import plotly.graph_objects as go
+    except ImportError:
+        raise ImportError("Plotly required for using plot()")
+    except RuntimeError:
```

Comment:
When do we think this will happen, is it e.g. running on a remote server? Should we test (i.e., just by trying, not writing tests) how this looks like?

Reply:
This is something we should test as well 😉

Reply:
@josh146, how would we test it? I guess this error might happen naturally on CI?

Reply:
Could you mock the import so that it causes a `RuntimeError`?